### PR TITLE
Interflux #0

### PR DESCRIPTION
**Pull request by noescape00**
This can be reviewed on commit by commit basis. 

Mostly refactoring, comments, minor fixes, setup for future adding chains other than ETH for interoperability.

### CODE REVIEW

File: src/Stratis.Features.FederatedPeg/Conversion/ConversionRequest.cs [hunk 1]
```diff
@@ -58,6 +67,9 @@ public class ConversionRequest : IBitcoinSerializable
         /// </summary>
         public string DestinationAddress { get { return this.destinationAddress; } set { this.destinationAddress = value; } }
 
+        /// <summary>Chain on which STRAX minting or burning should occur.</summary>
+        public DestinationChain DestinationChain { get { return (DestinationChain)this.destinationChain; } set { this.destinationChain = (int)value; } }
```
Comment (zeptin) [inline src/Stratis.Features.FederatedPeg/Conversion/ConversionRequest.cs:71]: If we are introducing this new field we will ideally need a migration mechanism for existing conversion requests stored by the multisig nodes.

Comment (noescape00) [reply src/Stratis.Features.FederatedPeg/Conversion/ConversionRequest.cs:71]: Given that we now have almost no conversion requests and that we can have multisig nodes upgraded in a few hours I think we can do without it at this stage. 

But I agree, migration mechanism will be needed later on

File: src/Stratis.Features.FederatedPeg/SourceChain/Deposit.cs [hunk 2]
```diff
@@ -12,6 +13,7 @@ public Deposit(uint256 id, DepositRetrievalType retrievalType, Money amount, str
             this.RetrievalType = retrievalType;
             this.Amount = amount;
             this.TargetAddress = targetAddress;
+            //this.TargetChain = targetChain;
```
Comment (zeptin) [inline src/Stratis.Features.FederatedPeg/SourceChain/Deposit.cs:16]: Is this going to be introduced later?

Comment (noescape00) [reply src/Stratis.Features.FederatedPeg/SourceChain/Deposit.cs:16]: Yes. Adding it in the constructor breaks a lot of tests, I was about to work on it but had to switch to other task, so disabled it in this PR.

File: src/Stratis.Features.FederatedPeg/TargetChain/MaturedBlocksSyncManager.cs [hunk 3]
```diff
@@ -19,6 +19,8 @@ namespace Stratis.Features.FederatedPeg.TargetChain
     /// Handles block syncing between gateways on 2 chains. This node will request
     /// blocks from another chain to look for cross chain deposit transactions.
     /// </summary>
+    /// <remarks>Processes matured block deposits from the cirrus chain and creates instances of <see cref="ConversionRequest"/> which are
+    /// saved to <see cref="IConversionRequestRepository"/>.</remarks>
```
Comment (zeptin) [inline src/Stratis.Features.FederatedPeg/TargetChain/MaturedBlocksSyncManager.cs:23]: Bear in mind that the creation of `ConversionRequest`s is something that was added onto this existing functionality. Conversion requests are obviously not used for normal cross-chain transfers.

**Review by zeptin — APPROVED** [review]

**Pull request merged**